### PR TITLE
Bump support up to requiring Java 8

### PR DESCRIPTION
**Pull request by camporter**
* Use built in base64 functionality.
* Use lambdas
* Use string based switches
* The API now requires TLS 1.2, java 8 is the first version that negotiates TLS 1.2 without any complications.

This will make it easier to add object filter support on top, and maybe also change from javawriter to it's new version, javapoet.

### CODE REVIEW

File: gen/src/main/java/com/softlayer/api/gen/MetaConverter.java [hunk 1]
```diff
@@ -76,13 +72,14 @@ public String getPackageName(String typeName) {
...
-            pkg.append('.').append(getValidJavaIdentifier(piece));
-        }
+        Arrays.stream(pieces)
+                .limit(pieces.length - 1) // Skip the last, it's the class name
+                .map(String::toLowerCase)
+                .map(piece -> {
+                    String replacement = keywordReplacements.get(piece);
+                    return replacement != null ? replacement : piece;
+                })
+                .forEach(piece -> pkg.append('.').append(getValidJavaIdentifier(piece)));
```
Comment (cretz) [inline gen/src/main/java/com/softlayer/api/gen/MetaConverter.java]: From one iteration to three...doesn't really matter, but strange to see the change just for change-sake

File: src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java [hunk 2]
```diff
@@ -10,4 +13,22 @@ public HttpBasicAuthCredentials(String username, String apiKey) {
...
+     *
+     * @return String
+     * @throws UnsupportedEncodingException If encoding with UTF-8 fails.
+     */
+    public String getHeader() throws UnsupportedEncodingException
+    {
+        String authPair = username + ':' + apiKey;
+        String result = "Basic ";
+        result += new String(
+            Base64.getEncoder().encode(authPair.getBytes("UTF-8")),
```
Comment (cretz) [inline src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java]: Inconsistent whether you choose `"UTF-8"` or `StandardCharsets.UTF_8`

File: src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java [hunk 3]
```diff
@@ -10,4 +13,22 @@ public HttpBasicAuthCredentials(String username, String apiKey) {
...
+     * for use in an HTTP Authorization header.
+     *
+     * @return String
+     * @throws UnsupportedEncodingException If encoding with UTF-8 fails.
+     */
+    public String getHeader() throws UnsupportedEncodingException
+    {
+        String authPair = username + ':' + apiKey;
+        String result = "Basic ";
+        result += new String(
```
Comment (cretz) [inline src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java]: Consider `Base64.getEncoder().encodeToString` (none of the chars are unicode, doesn't matter what the charset is on the way back)

File: src/test/java/com/softlayer/api/json/GsonJsonMarshallerFactoryTest.java [hunk 4]
```diff
@@ -128,7 +129,14 @@ public void testWrite() throws Exception {
...
         String regular = "\"1984-02-25T20:15:25-06:00\"";
-        Calendar expected = new GregorianCalendar(1984, Calendar.FEBRUARY, 25, 20, 15, 25);
+        Calendar expected = new GregorianCalendar(
+            1984,
+            Calendar.FEBRUARY,
+            25,
+            20,
+            15,
+            25
+        );
```
Comment (cretz) [inline src/test/java/com/softlayer/api/json/GsonJsonMarshallerFactoryTest.java]: What a great day in history!

File: gen/src/main/java/com/softlayer/api/gen/MetaConverter.java (hunk 1 above)
Comment (camporter) [reply gen/src/main/java/com/softlayer/api/gen/MetaConverter.java]: You caught me  :) this was my excuse to play around with streams. I'll revert this.

File: src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java (hunk 3 above)
Comment (camporter) [reply src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java]: Oh nice, much better

File: gen/src/main/java/com/softlayer/api/gen/MetaConverter.java (hunk 1 above)
Comment (camporter) [reply gen/src/main/java/com/softlayer/api/gen/MetaConverter.java]: Changed this back.

File: src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java (hunk 3 above)
Comment (camporter) [reply src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java]: Changed to use encodeToString and getBytes

File: src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java [hunk 5]
```diff
@@ -19,16 +18,10 @@ public HttpBasicAuthCredentials(String username, String apiKey) {
      * for use in an HTTP Authorization header.
      *
      * @return String
-     * @throws UnsupportedEncodingException If encoding with UTF-8 fails.
      */
-    public String getHeader() throws UnsupportedEncodingException
+    public String getHeader()
     {
```
Comment (cretz) [inline src/main/java/com/softlayer/api/http/HttpBasicAuthCredentials.java]: Brace placement inconsistency

File: src/test/java/com/softlayer/api/http/HttpBasicAuthCredentialsTest.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class HttpBasicAuthCredentialsTest {
+
+    @Test
+    public void testConstructor()
+    {
```
Comment (cretz) [inline src/test/java/com/softlayer/api/http/HttpBasicAuthCredentialsTest.java]: Brace placement inconsistency here and below

Comment (camporter) [reply src/test/java/com/softlayer/api/http/HttpBasicAuthCredentialsTest.java]: Bah! Need to add checkstyling to my editor to catch these...

**Pull request merged**